src/app.py

```diff
@@ -1,73 +1,354 @@
 
-db.init_app(app)
+CORS(app)
 
+db.init_app(app)
 with app.app_context():
     logger.info("Creating all tables...")
     db.create_all()
     logger.info("Created tables..")
 
+os.makedirs("tmp", exist_ok=True)
 
 logger.info("Server up and running...")
 
+# TODO: Need to decide on the approach of single jwt token / individual jwt token based on user credentails
+@app.route("/auth/login/", methods=["POST"])
+def login():
+    request_data = request.get_json()
+    if request_data["username"] and request_data["password"]:
+        logger.info("Generating JWT Token")
+        token = jwt.encode(
+            {
+                "user": request_data["username"],
+                "password": request_data["password"],
+                "exp": datetime.utcnow()
+                + timedelta(seconds=app.config["JWT_TOKEN_EXPIRATION_SECONDS"]),
+            },
+            app.config["SECRET_KEY"],
+            "HS256",
+        )
+        return jsonify({"token": token})
+    else:
+        abort_json(403, error="AUTHENTICATION_FAILED", message="Unable to verify")
 
-@app.route("/test", methods=["GET"])
-def test():
-    data = CatalogueItem.query.all()
-    print(data)
-    return "Catalogue server v1 api!"
 
+@app.route("/catalogue/<uuid>/", methods=["GET"])
+@token_required
+def get_catalogue(uuid: str):
+    """
+    GET single item from the database
+    """
+    logger.info("/catalogue/<uuid> GET called")
+    res = CatalogueItem.query.filter_by(uuid=uuid).first()
+    if not res:
+        abort_json(400, error="DATA_NOT_FOUND", message="Item not found!")
+    res = CatalogueItemSchema().dump(res)
+    return jsonify(res)
+
+
+@app.route("/catalogue/", methods=["GET"])
+@token_required
+def list_catalogue():
+    """
+    This API is used to select the CatalogueItem table based on query fitlers:
+        - transfer_status (reference: `services.db.enums.TransferStatus`)
+        - page (used for pagination)
+    """
+    logger.info("/catalogue/ - GET called")
+    status = (
+        request.args.get("transfer_status", TransferStatus.NOT_STARTED.value)
+        .strip()
+        .upper()
+    )
+
+    page = 1
+    try:
+        page = int(request.args.get("page", 1))
+    except:
+        logger.warning("Defaulting page to 1")
+        page = 1
+
+    logger.debug(f"page = {page}")
+    logger.debug(f"status = {status}")
+
+    res = CatalogueItem.query.filter_by(transfer_status=status).paginate(
+        page=page,
+        per_page=CFG.ITEMS_PER_PAGE,
+        error_out=False,
+    )
+    res = CatalogueItemSchema(many=True).dump(res.items)
+    logger.debug(f"Total rows selected = {len(res)}")
+
+    return jsonify(res)
+
+
+@app.route("/catalogue/", methods=["POST"])
+@token_required
+def create_catalogue():
+    """
+    Create single catalog item record and return the created record.
+    """
+    logger.info("/catalogue/ POST called")
+    data = request.json
+
+    mandatory_fields = ["name", "checksum_algorithm", "checksum_value"]
```

Comment:
Move to constants

Reply:
I have moved these to the configuration `CATALOGUE_POST_MANDATORY_FIELDS` as constant in the base config.